payu/models/cice.py

```diff
@@ -333,6 +342,47 @@ def archive(self, **kwargs):
         else:
             shutil.rmtree(self.work_input_path)
 
+        if self.compression_enabled():
+            self.compress_log_files()
+
+    def compression_enabled(self):
+        """
+        Determine whether to run log compression based on config.yaml settings.
+        Default to True when 'compress_logs' setting is absent.
+        """
+        archive_config = self.expt.config.get('archive', {})
+        if isinstance(archive_config, dict):
+            return archive_config.get('compress_logs', True)
+        else:
+            return True
+
+    def get_log_files(self):
+        """
+        Find model log files in the work directory based on regex patterns
+        in self.logs_to_compress.
+
+        Returns
+        -------
+        log_files: list of paths to model log files.
+        """
+        log_files = []
+        for filename in os.listdir(self.work_path):
+            if any((re.match(pattern, filename)
+                    for pattern in self.logs_to_compress)):
+                log_files.append(os.path.join(self.work_path, filename))
+        return log_files
+
+    def compress_log_files(self):
+        """
+        Compress model log files into tarball.
+        """
+        log_files = self.get_log_files()
+        with tarfile.open(name=os.path.join(self.work_path, self.log_tar_name),
+                          mode="w:gz") as tar:
+            for file in log_files:
+                tar.add(file, arcname=os.path.basename(file))
+                os.remove(file)
+
```

Comment:
```suggestion
        try:
            with tarfile.open(name=os.path.join(self.work_path, self.log_tar_name),
                              mode="w:gz") as tar:
                for file in log_files:
                   tar.add(file, arcname=os.path.basename(file))
        except:
            # catch some exception
        finally:
            for file in log_files: os.remove(file)
```
I think it might be better to separate the tar bit from the deletion. I know you've written tests for it, but it seems at some stage it [wasn't safe to remove the file until the tar object was written](https://stackoverflow.com/a/48292904/4727812). That may no longer be the case, or perhaps I misinterpreted the SO post.

Reply:
That's a good point. It seemed to be working reasonably but I haven't been able to find any more info on this since it was posted, so it's probably safest to delete the files after the writing is finished. 

I've just implemented this as:
https://github.com/ACCESS-NRI/payu/blob/d4e333473702f9938e73fce43c40855ae7ce08a8/payu/models/cice.py#L380-L387

and opted to just delete the files directly after the tarfile is written, so that nothing will be deleted if any exceptions do come up when writing the tarball. Let me know if this looks ok!